src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/Hover/ComponentAvailabilityService.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.AspNetCore.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Tooltip;
+using Microsoft.CodeAnalysis.Remote.Razor.ProjectSystem;
+
+namespace Microsoft.CodeAnalysis.Remote.Razor;
+
+internal sealed class ComponentAvailabilityService(RemoteSolutionSnapshot solutionSnapshot) : AbstractComponentAvailabilityService
```

Comment:
nit: Does this need to be in `Hover` anymore?

Reply:
Sorry, I don't understand the question. I might be tripping up on the word "anymore", since this is a new class.

If you're referring to the namespace, it's in the `Microsoft.CodeAnalysis.Remote.Razor` namespace to match the other file in the same folder.

Reply:
I apparently need some caffeine :laughing: 

Overall the `ComponentAvailabilityService` doesn't seem to be explicitly tied to `Hover` but it is in the `Hover` folder. Ignore the "anymore" part of the statement. That's just brain not braining.

Reply:
Gotcha! Nothing else currently uses this service and it's derived from implementation details of hover/tooltip. Its base implementation lives in `MS.CA.Razor.Workspaces\Tooltip` because it is implemented using hover/tooltip-specific extension methods that were in the same folder. It _could_ become a general service that is usable by all, but that's really outside the scope of this change. I wouldn't know the right place to put such a thing.